litmus-portal/frontend/src/views/CreateWorkflow/TuneWorkflow/TuneWorkflowSteps/TargetApplication.tsx

```diff
@@ -84,136 +178,365 @@ const TargetApplication: React.FC<TargetApplicationProp> = ({
+              <InputLabel id="appKind" style={{ color: '#00000099' }}>
+                {t('createWorkflow.tuneWorkflow.verticalStepper.appkind')}
+              </InputLabel>
+              <Select
+                labelId="appKind"
+                value={targetApp.appkind}
+                onChange={(event) => {
+                  setTargetApp({
+                    ...targetApp,
+                    appkind: event.target.value as string,
+                  });
+                  handleLabelChange();
+                  if (event.target.value === 'deployment') {
+                    setGVRObj({
+                      group: 'apps',
+                      version: 'v1',
+                      resource: 'deployments',
+                    });
+                  } else if (event.target.value === 'statefulset') {
+                    setGVRObj({
+                      group: 'apps',
+                      version: 'v1',
+                      resource: 'statefulsets',
+                    });
+                  } else if (event.target.value === 'daemonset') {
+                    setGVRObj({
+                      group: 'apps',
+                      version: 'v1',
+                      resource: 'daemonsets',
+                    });
+                  } else if (event.target.value === 'deploymentconfig') {
+                    setGVRObj({
+                      group: 'apps.openshift.io',
+                      version: 'v1',
+                      resource: 'deploymentconfigs',
+                    });
+                  } else if (event.target.value === 'rollout') {
+                    setGVRObj({
+                      group: 'argoproj.io',
+                      version: 'v1alpha1',
+                      resource: 'rollouts',
+                    });
+                  } else {
+                    setGVRObj({
+                      group: '',
+                      version: '',
+                      resource: '',
+                    });
+                  }
+                }}
+                label="appKind"
+              >
+                <MenuItem aria-label="None" value="" />
+                <MenuItem value="deployment">
+                  {t('createWorkflow.tuneWorkflow.verticalStepper.deployment')}
+                </MenuItem>
+                <MenuItem value="statefulset">
+                  {t('createWorkflow.tuneWorkflow.verticalStepper.statefulset')}
+                </MenuItem>
+                <MenuItem value="daemonset">
+                  {t('createWorkflow.tuneWorkflow.verticalStepper.daemonset')}
+                </MenuItem>
+                <MenuItem value="deploymentconfig">
+                  {t(
+                    'createWorkflow.tuneWorkflow.verticalStepper.deploymentconfig'
+                  )}
+                </MenuItem>
+                <MenuItem value="rollout">
+                  {t('createWorkflow.tuneWorkflow.verticalStepper.rollout')}
+                </MenuItem>
+              </Select>
```

Comment:
use constants for the value or use a map to dynamically generate these menu jsx it would help to scale in future.